src/library/dao/trackdao.cpp

```diff
@@ -1397,6 +1394,44 @@ TrackPointer TrackDAO::getTrackById(TrackId trackId) const {
     return pTrack;
 }
 
+TrackId TrackDAO::getTrackIdByRef(
+        const TrackRef& trackRef) const {
+    if (trackRef.getId().isValid()) {
+        return trackRef.getId();
+    }
+    {
+        GlobalTrackCacheLocker cacheLocker;
+        const auto track = cacheLocker.lookupTrackByRef(trackRef);
+        if (track) {
+            return track->getId();
+        }
+    }
+    return getTrackIdByLocation(trackRef.getLocation());
+}
+
+TrackPointer TrackDAO::getTrackByRef(
+        const TrackRef& trackRef) const {
+    if (!trackRef.isValid()) {
+        return TrackPointer();
+    }
+    {
+        GlobalTrackCacheLocker cacheLocker;
+        auto track = cacheLocker.lookupTrackByRef(trackRef);
+        if (track) {
+            return track;
+        }
+    }
```

Comment:
the code above is used in TrackDAO::getTrackIdByRef() as well. Common function?

Reply:
No, doesn't help. There are subtle differences and you could only extract a single line. I already tried it.

Reply:
OK